driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java

```diff
@@ -0,0 +1,589 @@
+            stmt.addBatch("insert into test (id) values (3)");
+            stmt.addBatch("insert into test (id) values (4)");
+            stmt.addBatch("insert into test (id) values (5)");
+
+            int[] results = stmt.executeBatch();
+            assertArrayEquals(results, new int[]{1, 1, 1});
+            conn.createStatement().execute("refresh table test");
+            ResultSet resultSet = conn.createStatement().executeQuery("select count(*) from test");
+            assertThat(resultSet.next(), is(true));
+            assertThat(resultSet.getLong(1), is(4L));
+        }
+    }
+
+    @Test
+    @Ignore("validate batch behaviour")
+    public void testExecuteBatchStatementFail() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            Statement stmt = conn.createStatement();
+            stmt.addBatch("insert into test (id) values (3)");
+            stmt.addBatch("insert into test (id) values (5)");
+            stmt.addBatch("select * from sys.cluster");
+
+            try {
+                stmt.executeBatch();
+                fail("BatchUpdateException not thrown");
+            } catch (BatchUpdateException e) {
+                assertArrayEquals(e.getUpdateCounts(), new int[]{1, 1, Statement.EXECUTE_FAILED});
+            }
+            conn.createStatement().execute("refresh table test");
+            ResultSet resultSet = conn.createStatement().executeQuery("select count(*) from test");
+            assertThat(resultSet.next(), is(true));
+            assertThat(resultSet.getLong(1), is(3L));
+        }
+    }
+
+    @Test
+    public void testExecuteBatchPreparedStatement() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            PreparedStatement stmt = conn.prepareStatement("insert into test (id, string_field) values (?, ?)");
+            stmt.setInt(1, 2);
+            stmt.setString(2, "foo");
+            stmt.addBatch();
+
+            stmt.setInt(1, 3);
+            stmt.setString(2, "bar");
+            stmt.addBatch();
+
+            stmt.setInt(1, 4);
+            stmt.setString(2, "baz");
+            stmt.addBatch();
+
+            int[] results = stmt.executeBatch();
+            assertArrayEquals(new int[]{1, 1, 1}, results);
+            conn.createStatement().execute("refresh table test");
+            ResultSet resultSet = conn.createStatement().executeQuery("select count(*) from test");
+            assertThat(resultSet.next(), is(true));
+            assertThat(resultSet.getLong(1), is(4L));
+        }
+    }
+
+    @Test
+    public void testExecuteBatchPreparedStatementFailBulkTypes() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionStr)) {
+            PreparedStatement stmt = conn.prepareStatement("insert into test (id, string_field) values (?, ?)");
+            stmt.setInt(1, 2);
+            stmt.setString(2, "foo");
```

Comment:
also here using a `id` column should be enough, cast failure could be done on that column as well afaik